sstables/sstables.cc

```diff
@@ -518,16 +524,22 @@ future<> parse(const schema& schema, sstable_version_types v, random_access_read
                      s.header.memory_size,
                      s.header.sampling_level,
                      s.header.size_at_full_sampling);
-    auto buf = co_await in.read_exactly(s.header.size * sizeof(pos_type));
     auto len = s.header.size * sizeof(pos_type);
-    check_buf_size(buf, len);
+    // Read using fragmented buffer to avoid large contiguous allocations
+    auto frag_buf = co_await in.read_exactly_fragmented(len);
+    check_fragmented_buf_size(frag_buf, len);
 
-    // Positions are encoded in little-endian.
-    auto b = buf.get();
+    static_assert(fragmented_temporary_buffer::default_fragment_size % sizeof(pos_type) == 0,
+        "parse() assumes that the default fragment size (currently 128 KiB) is a multiple of pos_type");
     s.positions.reserve(s.header.size + 1);
-    while (s.positions.size() != s.header.size) {
-        s.positions.push_back(seastar::read_le<pos_type>(b));
-        b += sizeof(pos_type);
+    for (auto& buf : frag_buf) {
+        auto b = buf.get();
```

Comment:
It's simpler to iterate over `in.read_exactly(sizeof(pos_type))`, there's no need to pre-read the entire thing into memory.

input_stream uses 128kB buffers.